### PR TITLE
groupBy friendly tagname fix

### PR DESCRIPTION
**Pull request by gsps1**
https://issues.cask.co/browse/CDAP-1850

build running : http://builds.cask.co/browse/CDAP-RBT187

### CODE REVIEW

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java [hunk 1]
```diff
@@ -267,44 +268,34 @@ public void testQueryMetrics() throws Exception {
...
   }
 
   private void verifyGroupByResult(String url, List<TimeSeriesResult> groupByResult) throws Exception {
     MetricQueryResult result = post(url, MetricQueryResult.class);
     Assert.assertEquals(groupByResult.size(), result.getSeries().length);
     for (MetricQueryResult.TimeSeries timeSeries : result.getSeries()) {
+      boolean timeSeriesMatchFound = false;
       for (TimeSeriesResult expectedTs : groupByResult) {
-        if (compareTagValues(expectedTs.getTagValues(), timeSeries.getGrouping())) {
+        if (expectedTs.getTagValues().equals(ImmutableMap.copyOf(timeSeries.getGrouping()))) {
```
Comment (abaranau) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: why do you need to copy a map?

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java [hunk 2]
```diff
@@ -318,11 +327,19 @@ private MetricQueryResult decorate(Collection<MetricTimeSeries> series, long sta
     for (MetricTimeSeries timeSeries : series) {
       MetricQueryResult.TimeValue[] timeValues = decorate(timeSeries.getTimeValues());
       serieses[i++] = new MetricQueryResult.TimeSeries(timeSeries.getMetricName(),
-                                                       timeSeries.getTagValues(), timeValues);
+                                                       replaceTagNames(timeSeries.getTagValues()), timeValues);
     }
     return new MetricQueryResult(startTs, endTs, serieses);
   }
 
+  private Map<String, String> replaceTagNames(Map<String, String> tagValues) {
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java]: can we call it tagNamesToHuman

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java [hunk 3]
```diff
@@ -170,8 +171,16 @@ public void query(HttpRequest request, HttpResponder responder,
 
   private List<String> parseGroupBy(String groupBy) {
     // groupBy tags are comma separated
-    return (groupBy == null) ? Lists.<String>newArrayList() :
+    List<String> groupByTags = (groupBy == null) ? Lists.<String>newArrayList() :
       Lists.newArrayList(Splitter.on(",").split(groupBy).iterator());
+    groupByTags = Lists.transform(groupByTags, new Function<String, String>() {
```
Comment (abaranau) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/query/MetricsHandler.java]: extract into humanToTagNames?

**Comment by abaranau** [comment]
minor comments

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java (hunk 1 above)
Comment (gsps1) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsHandlerTestRun.java]: so we can remove compareTagValues function

**Comment by abaranau** [comment]
@gsps1 can we merge?

**Pull request merged**